### PR TITLE
Jungfrau features: 4. defaultdacs

### PR DESCRIPTION
**Pull request by thattil**
(for all chips)
    • set default dac value (will not set dac to that value) (sett = undefined sent to detector)
    • if setting given as argument, then set default for that setting (will not set dac to that value). Only for special dacs.
  
M3:
  • similar way of Jungfrau in setting special dacs

Todo:
python

### CODE REVIEW

File: slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c [hunk 1]
```diff
@@ -1201,59 +1284,74 @@ int getAllTrimbits() {
...
-    } else if (detectorDacs[M_VRPREAMP] == DEFAULT_FAST_VRPREAMP &&
-               detectorDacs[M_VRSHAPER] == DEFAULT_FAST_VRSHAPER) {
-        if (thisSettings != FAST) {
-            thisSettings = FAST;
-            LOG(logINFOBLUE, ("Validated Settings changed to fast!\n"));
+    // if any special dac value is changed individually => undefined
+    const int specialDacs[NSPECIALDACS] = SPECIALDACINDEX;
+    int *specialDacValues[] = {defaultDacValue_standard, defaultDacValue_fast,
+                               defaultDacValue_highgain};
+    int settList[NUMSETTINGS] = {STANDARD, FAST, HIGHGAIN};
```
Comment (erikfrojdh) [inline slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c:1324]: Maybe remove numsettings to be consistent with the line above?

**Review by erikfrojdh — COMMENTED** [review]
Small comments, otherwise good. Need make sure we have tests that catches the logic.

File: slsDetectorSoftware/include/sls/Detector.h [hunk 2]
```diff
@@ -1779,6 +1794,10 @@ class Detector {
     void updateRxRateCorrections();
     void setNumberofUDPInterfaces_(int n, Positions pos);
     Result<int> getNumberofUDPInterfaces_(Positions pos) const;
+    Result<int> getDefaultDac_(defs::dacIndex index,
+                               defs::detectorSettings sett, Positions pos = {});
+    void setDefaultDac_(defs::dacIndex index, int defaultValue,
```
[the remark this answers is not in the document]
Comment (thattil) [reply slsDetectorSoftware/include/sls/Detector.h:1799]: makes sense. will look into it.

File: slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c (hunk 1 above)
Comment (thattil) [reply slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c:1324]: done

File: slsDetectorSoftware/include/sls/Detector.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (thattil) [reply slsDetectorSoftware/include/sls/Detector.h]: donr, except for setNumberofudpinterfaces_. The code does not look user friendly afterwards.

**Pull request merged**